otelcol/collector.go

```
@@ -179,6 +180,9 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	}
 
 	if err = cfg.Validate(); err != nil {
+		if errors.Is(err, errInvalidOrMissingConfigFile) {
+			return errInvalidOrMissingConfigFile
+		}
```

```suggestion
```

We can return the error directly.

---

Okay